### PR TITLE
XEB fidelity estimator based on normalized HOG score

### PR DESCRIPTION
**Pull request by viathor**

### CODE REVIEW

**Comment by viathor** [comment]
Tests failed due to #2368. The relevant test in fidelity_estimation_test.py does pass.

**Comment by viathor** [comment]
@yzniu This is the estimator we discussed.

File: cirq/experiments/fidelity_estimation.py [hunk 1]
```diff
@@ -99,6 +99,41 @@ def log_xeb_fidelity_from_probabilities(
...
+
+    See `linear_xeb_fidelity_from_probabilities` for the assumptions made
+    by this estimator.
+
+    The mean of this estimator is the true fidelity f and the variance is
+
+        (1/log(2)^2 - f^2) / M
+
+    where f is the fidelity and M the number of observations, equal to
+    len(probabilities). This is always worse than log XEB (see above).
```
Comment (kevinsung) [inline cirq/experiments/fidelity_estimation.py]: If it's always worse, why would someone want to use it?

Comment (viathor) [reply cirq/experiments/fidelity_estimation.py]: One use-case I encountered recently: plotting various estimators for comparison. A reason to include HOG-based estimates on that plot was that HOG test is fairly well-known (and variance was negligible for all of them anyway).

File: cirq/experiments/fidelity_estimation.py [hunk 2]
```diff
@@ -99,6 +99,41 @@ def log_xeb_fidelity_from_probabilities(
             np.mean(np.log(probabilities)))
 
 
+def hog_score_xeb_fidelity_from_probabilities(
+        hilbert_space_dimension: int,
+        probabilities: Sequence[float],
+) -> float:
+    """XEB fidelity estimator based on normalized HOG score.
```
Comment (Strilanc) [inline cirq/experiments/fidelity_estimation.py]: Add a reference that defines the HOG score.

Comment (viathor) [reply cirq/experiments/fidelity_estimation.py]: Done.

**Pull request merged**